src/mysql_protocol.erl

```diff
@@ -1220,6 +1214,86 @@ recv_packet(SockModule, Socket, Timeout, ExpectSeqNum, Acc) ->
             {error, Reason}
     end.
 
+-spec send_file(module(), term(), Filename :: binary(), AllowedPaths :: [binary()],
+                SeqNum :: integer()) ->
+    {ok | {error, term()}, NextSeqNum :: integer()}.
+send_file(SockModule, Socket, Filename, AllowedPaths, SeqNum0) ->
+    {Result, SeqNum1} = case allowed_path(Filename, AllowedPaths) andalso
+                             file:open(Filename, [read, raw, binary]) of
+        false ->
+            {{error, not_allowed}, SeqNum0};
+        {ok, Handle} ->
+            {ok, SeqNum2} = send_file_chunk(SockModule, Socket, Handle, SeqNum0),
+            ok = file:close(Handle),
+            {ok, SeqNum2};
+        {error, _Reason} = E ->
+            {E, SeqNum0}
+    end,
+    {ok, SeqNum3} = send_packet(SockModule, Socket, <<>>, SeqNum1),
+    {Result, SeqNum3}.
+
+-spec allowed_path(binary(), [binary()]) -> boolean().
+allowed_path(Path, AllowedPaths) ->
+    valid_path(Path) andalso
+    binary:last(Path) =/= $/ andalso
+    lists:any(
+        fun
+            (AllowedPath) when Path =:= AllowedPath ->
+                true;
+            (AllowedPath) when binary_part(Path, 0, byte_size(AllowedPath)) =:= AllowedPath ->
+                binary:last(AllowedPath) =:= $/;
```

Comment:
I think it's a bit harsh to require directories to he specified with a trailing slash. It's not required by most unix commands and applications in general. It wouldn't be too hard to add something like a fun clause for allowed directory without trailing slash: `(<<AllowedPath:Size/binary, "/", _/binary>>) when AllowedPath doesn't end with a slash`.

Reply:
Ok, I changed as suggested, to work with allowed paths with and without slashes.